### DISCUSSION TITLE
Adding dependencies to a Machine

### DISCUSSION THREAD

**Discussion by wowcodeisgood** [Eomys/pyleecan#697] [Q&A]
Hello I am new here (bloody beginner), have my troubles to make my machine parametric want to make a SIPMSM machine. So right now I have no controll over the geometry, have tried to make help varibles and call the Class stator.slot.W0 for example and then make the dependencies with other class values like the rotor that kind of worked but I get errors. Help is appreciated, thanks:)

machine = load(join(DATA_DIR, "Machine", "SPMSM_001.json"))

mm = 1e-3
# Main geometry parameter

air_gap = 1*mm
magnet_height = 5*mm
yoke_thickness = 5* mm  # Thickness of the yoke

stator = LamSlotWind(
    Rint= 40, # internal radius [m]
    Rext= 45 * mm, # external radius [m]
    L1=50 * mm, # Lamination stack active length [m] without radial ventilation airducts
                # but including insulation layers between lamination sheets
    Nrvd=0, # Number of radial air ventilation duct
    Kf1=0.95, # Lamination stacking / packing factor
    is_internal=False,
    is_stator=True,
)

stator.winding = Winding(qs=3, p=2, Nlayer=2)
# External Rotor
stator.winding.conductor = CondType12(Wwire=0.5*mm , Wins_cond=0.5*mm, Nwppc=1, Wins_wire=0,Kwoh=0.5)

CMT_Magnet = Material(name="CMT_Magnet")

M400_50A = load(join(DATA_DIR, "Material", "M400-50A.json"))
Copper1 = load(join(DATA_DIR, "Material", "Copper1.json"))

# Slot setup
stator.slot = SlotW15(
    
    W0=3*mm,
    W3=3*mm,
    H0=2*mm,
    H1=3*mm,
    H2=10*mm,
    R1= 1*mm,
    R2=0.0000001*mm,
    Zs=12
)
stator.Rint = stator.Rext-stator.slot.H2 - stator.slot.H1 -stator.slot.H0 - yoke_thickness 
stator.slot.R1=stator.slot.H1*0.1

rotor = LamSlotMag(Rext=stator.Rint-air_gap-magnet_height, Rint=0, is_internal=True, is_stator=False)

rotor.slot = SlotM18(magnet_height, Zs=6)

shaft = Shaft(Drsh=0, # Diamater of the rotor shaft [m]
                                   # used to estimate bearing diameter for friction losses
              Lshaft=50*mm # length of the rotor shaft [m]
             )
frame = None

**Comment by SebGue** [comment]
EDIT: Moved from issues (#695) to discussions, since this seem to be no Pyleecan issue for now.

Hello @wowcodeisgood ,

although I don't know exactly what you try to achieve, I get a machine plotted with your parameters.
There are only some missing multiply symbols, e.g. with Wwire and you forgot mm unit with Rint.
You'll find my corrections below. Hope this helps. 

Best regards, Sebastian

![grafik](https://github.com/Eomys/pyleecan/assets/43440731/179bdc8c-2964-4b87-9d09-7b9bb5a136c3)

```
machine = load(join(DATA_DIR, "Machine", "SPMSM_001.json"))

mm = 1e-3

### Main geometry parameter

air_gap = 1 * mm
magnet_height = 5 * mm
yoke_thickness = 5 * mm  # Thickness of the yoke

stator = LamSlotWind(
    Rint=40 * mm,  # internal radius [m]
    Rext=45 * mm,  # external radius [m]
    L1=50
    * mm,  # Lamination stack active length [m] without radial ventilation airducts
    # but including insulation layers between lamination sheets
    Nrvd=0,  # Number of radial air ventilation duct
    Kf1=0.95,  # Lamination stacking / packing factor
    is_internal=False,
    is_stator=True,
)

stator.winding = Winding(qs=3, p=2, Nlayer=2)

### External Rotor

stator.winding.conductor = CondType12(
    Wwire=0.5 * mm, Wins_cond=0.5 * mm, Nwppc=1, Wins_wire=0, Kwoh=0.5
)

CMT_Magnet = Material(name="CMT_Magnet")

M400_50A = load(join(DATA_DIR, "Material", "M400-50A.json"))
Copper1 = load(join(DATA_DIR, "Material", "Copper1.json"))

### Slot setup

stator.slot = SlotW15(
    W0=3 * mm,
    W3=3 * mm,
    H0=2 * mm,
    H1=3 * mm,
    H2=10 * mm,
    R1=1 * mm,
    R2=0.0000001 * mm,
    Zs=12,
)

stator.Rint = (
    stator.Rext - stator.slot.H2 - stator.slot.H1 - stator.slot.H0 - yoke_thickness
)
stator.slot.R1 = stator.slot.H1 * 0.1

rotor = LamSlotMag(
    Rext=stator.Rint - air_gap - magnet_height,
    Rint=0,
    is_internal=True,
    is_stator=False,
)

rotor.slot = SlotM18(magnet_height, Zs=6)

shaft = Shaft(
    Drsh=0,  # Diamater of the rotor shaft [m]
    # used to estimate bearing diameter for friction losses
    Lshaft=50 * mm,  # length of the rotor shaft [m]
)
frame = None

machine.plot()

```
